### PR TITLE
Environment for every PR

### PR DESCRIPTION
**Pull request by waveywaves**
## Change Summary

Provide summary of changes with issue number if any.

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

### CODE REVIEW

File: .github/workflows/uffizzi-build.yml [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    needs: 
+      - build-nocodb
+    outputs:
+      compose-file-cache-key:  ${{ env.COMPOSE_FILE_HASH }}
+    steps:
+      - name: Checkout git repo
+        uses: actions/checkout@v3
+      - name: Render Compose File
+        run: |
+          NOCODB_IMAGE=$(echo ${{ needs.build-nocodb.outputs.tags }})
```
Comment (axisofentropy) [inline .github/workflows/uffizzi-build.yml]: Does this need the `echo`? Should it work with just the substitution?

File: .github/workflows/uffizzi-preview.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        run: |
+          echo "PR number: ${{ env.PR_NUMBER }}"
+          echo "Compose file hash: ${{ env.COMPOSE_FILE_HASH }}"
+          cat event.json
+
+  deploy-uffizzi-preview:
+    name: Use Remote Workflow to Preview on Uffizzi
+    needs:
+      - cache-compose-file
+    uses: UffizziCloud/preview-action/.github/workflows/reusable.yaml@v2
```
Comment (axisofentropy) [inline .github/workflows/uffizzi-preview.yml:77]: `@v2.6.1`!

**Comment by axisofentropy** [comment]
This looks good. Great work integrating with their existing workflow.

I like that they specified `ready_for_review` as an event trigger, and they specified `paths`. We should consider adding these in our examples. https://github.com/waveywaves/nocodb/blob/d5e67e7abd53c9504cdbbbb6b6402ff4c28abc9f/.github/workflows/release-pr.yml#L9-L10

File: .github/workflows/uffizzi-preview.yml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    with:
+      # If this workflow was triggered by a PR close event, cache-key will be an empty string
+      # and this reusable workflow will delete the preview deployment.
+      compose-file-cache-key: ${{ needs.cache-compose-file.outputs.compose-file-cache-key }}
+      compose-file-cache-path: docker-compose.rendered.yml
+      server: https://app.uffizzi.com/
```
Comment (axisofentropy) [inline .github/workflows/uffizzi-preview.yml]: Remove trailing `/`.

File: .github/uffizzi/docker-compose.uffizzi.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    restart: always
+    environment:
+      POSTGRES_PASSWORD: password
+      POSTGRES_USER: postgres
+      POSTGRES_DB: root_db
+  nocodb:
+    image: "${NOCODB_IMAGE}"
+    ports:
+      - "8080:8080"
+    restart: always
```
Comment (axisofentropy) [inline .github/uffizzi/docker-compose.uffizzi.yml:34]: I don't think `ports` or `restart` are used by Uffizzi.

File: .github/workflows/release-pr.yml [hunk 5]
```diff
@@ -94,9 +95,46 @@ jobs:
             docker run -d -p 8888:8080 nocodb/nocodb-timely:${{ needs.set-tag.outputs.current_version }}-${{ needs.set-tag.outputs.target_tag }}
             ```
 
+  # Create a preview for the pull request
+  preview-pull-request:
+    name: "Trigger Uffizzi Preview"
+    if: ${{ github.event.pull_request.head.repo.full_name == github.repository && github.actor != 'dependabot[bot]' && github.event.pull_request.draft == false && github.base_ref == 'develop' && github.event.action != 'closed' }}
```
Comment (axisofentropy) [inline .github/workflows/release-pr.yml:101]: I like these conditions and we should probably make most of them standard.

File: .github/workflows/release-pr.yml [hunk 6]
```diff
@@ -94,9 +95,46 @@ jobs:
...
+        uses: actions/upload-artifact@v3
+        with:
+          name: preview-spec 
+          path: docker-compose.rendered.yml
+          retention-days: 2
+      - name: Serialize PR Event to File
+        run:  |
+          cat << EOF > event.json
+          ${{ toJSON(github.event) }} 
+
```
Comment (axisofentropy) [inline .github/workflows/release-pr.yml]: We don't want this empty line in the JSON file.

File: .github/workflows/release-pr.yml (hunk 5 above)
Comment (waveywaves) [reply .github/workflows/release-pr.yml:101]: I agree

File: .github/workflows/release-pr.yml (hunk 6 above)
Comment (waveywaves) [reply .github/workflows/release-pr.yml]: removing it

File: .github/uffizzi/docker-compose.uffizzi.yml (hunk 4 above)
Comment (waveywaves) [reply .github/uffizzi/docker-compose.uffizzi.yml:34]: keeping them for resusability with docker-compose

**Pull request merged**